packages/core/src/common/uri.ts

```diff
@@ -167,6 +167,13 @@ export default class URI {
         return this.withFragment('');
     }
 
+    /**
+     * return a new URI replacing the current with the given file
+     */
+    withFile(file: string): URI {
```

Comment:
I don't think it should be on `URI` to discourage creating/parsing file URIs on the frontend. Debug adapters are special case. How about moving `new URI(Uri.file(file))` to the debug extension?

It won't produce a normalized URI for the case when backend is Windows, but frontend is Linux, for instance
`\\home\\file.ts` path won't be transformed to `file:///home/file.ts`. In practice i have not seen such combination, so maybe it is fine for now.

cc @svenefftinge

Reply:
Hmm, I've just had this scenario working (backend Windows, frontend Mac).

I figured the `withFile()` function could be useful elsewhere and wanted to keep related functionality together. Would you prefer it was removed and I simply called `return new URI(Uri.file(raw.path));` instead?

Reply:
@akosyakov I've removed the `withFile()` function